security/security-design/shared-assets/kms-backup/files/kms_key_backup.py

```diff
@@ -0,0 +1,489 @@
+def overwrite_conf(args,config):
+    """ overwrite_conf """
+    dargs=vars(args)
+    for arg in dargs.keys():
+        if dargs[arg] is not None:
+            config[arg]=dargs[arg]
+    if 'exportonly' not in dargs or dargs['exportonly'] is None or not dargs['exportonly']:
+        config['exportonly']=False
+    else:
+        config['exportonly']=True
+
+#########################################################################
+# verify_config
+# Verifies that all mandatory parameters are set
+# Take the two modi emxortkey or copykey into account
+#########################################################################
+def verify_config(config):
+    """ verify_config, verifies that all mandatory configuration parameters is set """
+    #
+    # Mandatory source parameters
+    # source_keyname" or"source_key_ocid"checked for explisitt
+    #
+    mandatory_source_parameters = [ 'ociconfig','source_ociprofile','source_region','source_compartment','source_vault']
+    #
+    # Mandatory target parameters
+    #
+    mandatory_target_parameters=['target_region','target_compartment','target_vault','target_ociprofile', 'target_keyname']
+    #
+    # Mandatory target parameters
+    #
+    mandatory_other_parameters=['wrapping_algorithm']
+    #
+    complete=0
+    for param in mandatory_source_parameters:
+        if config[param] is None:
+            print("Mandatory source parameter "+param+ " is missing in configuration/commandline")
+            complete+=1
+    for param in mandatory_other_parameters:
+        if config[param] is None:
+            print("Mandatory other parameter "+param+ " is missing in configuration/commandline")
+            complete+=1
+    #
+    # Add default for non mandatory parameters
+    #
+    config['source_key_version'] = None if 'source_key_version' not in config else config['source_key_version']
+    config['source_keyname'] = None if 'source_keyname' not in config else config['source_keyname']
+    config['source_key_ocid'] = None if 'source_key_ocid' not in config else config['source_key_ocid']
+    #
+    # Explisitt check for "source_keyname" or "source_key_ocid" at least one should be set,
+    # if both, OCID takes presidents
+    #
+    if config['source_keyname'] is None and config['source_key_ocid'] is None :
+        print("Neither source_keyname nor source_key_ocid is set")
+        print(config)
+        complete+=1
+    #
+    # Only validate target config parameters if exportonly is not set
+    #
+    if not config['exportonly']:
+        #
+        # Include target verification
+        for param in mandatory_target_parameters:
+            if config[param] is None:
+                print("Mandatory target parameter "+param+ " is missing in configuration/commandline")
+                complete+=1
+    else:
+        if not 'wrapping_pubkey_file' in config:
+            print("wrapping_pubkey_file is required for exportonly option")
+            complete+=1 
+    return(complete == 0)

```

Comment:
No parentheses needed.

Reply:
Keeping it it, personal taste, makes FMPOV, makes it more clear